core/src/io/anuke/mindustry/world/blocks/power/PowerDiode.java

```diff
@@ -0,0 +1,86 @@
+package io.anuke.mindustry.world.blocks.power;
+
+import io.anuke.arc.Core;
+import io.anuke.mindustry.ui.Bar;
+import io.anuke.arc.util.Eachable;
+import io.anuke.mindustry.ui.Cicon;
+import io.anuke.mindustry.world.Tile;
+import io.anuke.mindustry.world.Block;
+import io.anuke.arc.graphics.g2d.Draw;
+import io.anuke.mindustry.graphics.Pal;
+import io.anuke.arc.graphics.g2d.TextureRegion;
+import io.anuke.mindustry.entities.traits.BuilderTrait;
+
+public class PowerDiode extends Block{
+    protected TextureRegion arrow;
+
+    public PowerDiode(String name){
+        super(name);
+        rotate = true;
+        update = true;
+        solid = true;
+        insulated = true;
+    }
+
+    @Override
+    public void update(Tile tile){
+        super.update(tile);
+
+        if(!tile.back().block().hasPower || !tile.front().block().hasPower) return;
+
+        PowerGraph backGraph = tile.back().entity.power.graph;
+        PowerGraph frontGraph = tile.front().entity.power.graph;
+        if(backGraph == frontGraph) return;
+
+        // 0f - 1f of battery capacity in use
+        float backStored = backGraph.getBatteryStored() / backGraph.getTotalBatteryCapacity();
+        float frontStored = frontGraph.getBatteryStored() / frontGraph.getTotalBatteryCapacity();
+
+        // try to send if the back side has more % capacity stored than the front side
+        if(backStored > frontStored) {
+            // send half of the difference
+            float amount = backGraph.getBatteryStored() * (backStored - frontStored) / 2;
+
+            backGraph.useBatteries(amount);
```

Comment:
A possible issue that may arise:
Graph 1 is transferring power to graph 2. The power fractions are as follows:

1: 1.0 - 50000 / 50000
2: 0.5 - 5/10

This would transfer (1.0 - 0.5) / 2 = 0.25 of the first graph, or 12500 power. For at least one frame, that's going to transfer a lot more power than the second graph can handle. A solution would be to `clamp(amount, 0, frontGraph.getTotalBatteryCapacity() * (1 - frontStored))` so no extra power is transferred.